js/filter.js

```diff
@@ -52,7 +52,9 @@ function populateSpeciesInfo(results, speciesID, mddTable) {
       if (document.location.hash != "") {
           var permalink = document.URL;            
       } else {
-          var permalink = document.URL + "genus=" + speciesData.genus + "&" + "species=" + speciesData.specificEpithet + "&" + "id=" + speciesID;
+          // The current URL may or may not have a "#" at the end already. Handle both cases.
```

Comment:
Nice! Why not use the new URL API https://developer.mozilla.org/en-US/docs/Web/API/URL#12 ?

Reply:
Partly to avoid making too many changes at once, partly because I don't think what we're doing here (URL parameters but in the hash part of the URL) is what the standard APIs are designed for. I do agree that it's better to use modern APIs like that instead of treating URLs as raw strings.

Reply:
I'll add some unit test for this . . . looks like a perfect case for some tests.

Reply:
Unit tests would be good, yes. I haven't looked at this codebase enough to see whether that would be easy to do.

Reply:
some permalink unit tests can be found at https://github.com/mammaldiversity/mammaldiversity.github.io/tree/master/test .